### PR TITLE
Add specific help command

### LINKED ISSUE #55: Add help command
Description can be found in our feature specification

### PR DESCRIPTION
**Pull request by jannaleong**
First-time users may be confused when
first using the application.

Specific help commands help users
easily find help for each specific command.

Create separate classes for each type of
command and a corresponding help window.

Fixes #55

### CODE REVIEW

File: src/main/java/seedu/address/ui/HelpEditWindow.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.input.Clipboard;
+import javafx.scene.input.ClipboardContent;
+import javafx.stage.Stage;
+import seedu.address.commons.core.LogsCenter;
+
+/**
+ * Controller for a help page
```
Comment (yleeyilin) [inline src/main/java/seedu/address/ui/HelpEditWindow.java:14]: I think this javadocs needs a full stop at the end

File: src/main/java/seedu/address/ui/HelpPoochMaintenanceWindow.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.input.Clipboard;
+import javafx.scene.input.ClipboardContent;
+import javafx.stage.Stage;
+import seedu.address.commons.core.LogsCenter;
+
+/**
+ * Controller for a help page
```
Comment (yleeyilin) [inline src/main/java/seedu/address/ui/HelpPoochMaintenanceWindow.java:14]: Same here

**Review by yleeyilin — APPROVED** [review]
LGTM! with some minor optional revision suggestions

File: src/main/java/seedu/address/ui/HelpDeleteWindow.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+     *         <li>
+     *             if this method is called on the primary stage.
+     *         </li>
+     *         <li>
+     *             if {@code dialogStage} is already showing.
+     *         </li>
+     *     </ul>
+     */
+    public void show() {
+        logger.fine("Showing help page about the application.");
```
Comment (jamessinmaojun) [inline src/main/java/seedu/address/ui/HelpDeleteWindow.java:70]: Maybe abstract the magic string out as a constant variable?

File: src/main/java/seedu/address/ui/HelpPoochMaintenanceWindow.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+     *         <li>
+     *             if this method is called on the primary stage.
+     *         </li>
+     *         <li>
+     *             if {@code dialogStage} is already showing.
+     *         </li>
+     *     </ul>
+     */
+    public void show() {
+        logger.fine("Showing help page about the application.");
```
Comment (jamessinmaojun) [inline src/main/java/seedu/address/ui/HelpPoochMaintenanceWindow.java:71]: same here, magic string.

File: src/main/java/seedu/address/ui/HelpPoochStaffWindow.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+     *         <li>
+     *             if this method is called on the primary stage.
+     *         </li>
+     *         <li>
+     *             if {@code dialogStage} is already showing.
+     *         </li>
+     *     </ul>
+     */
+    public void show() {
+        logger.fine("Showing help page about the application.");
```
Comment (jamessinmaojun) [inline src/main/java/seedu/address/ui/HelpPoochStaffWindow.java:71]: same here

File: src/main/java/seedu/address/ui/HelpPoochSupplierWindow.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+     *         <li>
+     *             if this method is called on the primary stage.
+     *         </li>
+     *         <li>
+     *             if {@code dialogStage} is already showing.
+     *         </li>
+     *     </ul>
+     */
+    public void show() {
+        logger.fine("Showing help page about the application.");
```
Comment (jamessinmaojun) [inline src/main/java/seedu/address/ui/HelpPoochSupplierWindow.java:71]: same here

File: src/main/java/seedu/address/ui/HelpSearchWindow.java [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+     *         <li>
+     *             if this method is called on the primary stage.
+     *         </li>
+     *         <li>
+     *             if {@code dialogStage} is already showing.
+     *         </li>
+     *     </ul>
+     */
+    public void show() {
+        logger.fine("Showing help page about the application.");
```
Comment (jamessinmaojun) [inline src/main/java/seedu/address/ui/HelpSearchWindow.java:77]: same here

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 8]
```diff
@@ -178,10 +245,28 @@ private CommandResult executeCommand(String commandText) throws CommandException
             logger.info("Result: " + commandResult.getFeedbackToUser());
             resultDisplay.setFeedbackToUser(commandResult.getFeedbackToUser());
 
-            if (commandResult.isShowHelp()) {
+            if (commandResult.isShowHelp() && commandResult.getFeedbackToUser()
```
Comment (jamessinmaojun) [inline src/main/java/seedu/address/ui/MainWindow.java:248]: I know you split the if statement condition into two lines but the condition is still rather verbose and might warrant an abstraction into a boolean variable

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 9]
```diff
@@ -178,10 +245,28 @@ private CommandResult executeCommand(String commandText) throws CommandException
...
+                    .equals(HelpCommand.SHOWING_HELP_MESSAGE)) {
                 handleHelp();
+            } else if (commandResult.isShowHelp() && commandResult.getFeedbackToUser()
```
Comment (jamessinmaojun) [inline src/main/java/seedu/address/ui/MainWindow.java:251]: same here, for the rest as well

**Review by jamessinmaojun — COMMENTED** [review]
lgtm! Just some comments about semantics but overall gj!

**Review by jamessinmaojun — APPROVED** [review]

File: src/main/java/seedu/address/ui/HelpEditWindow.java (hunk 1 above)
Comment (jannaleong) [reply src/main/java/seedu/address/ui/HelpEditWindow.java]: yes i agree!! i have added them

File: src/main/java/seedu/address/ui/HelpDeleteWindow.java (hunk 3 above)
Comment (jannaleong) [reply src/main/java/seedu/address/ui/HelpDeleteWindow.java]: i agree!! i have added a variable

File: src/main/java/seedu/address/ui/MainWindow.java (hunk 8 above)
Comment (jannaleong) [reply src/main/java/seedu/address/ui/MainWindow.java]: yes it is much neater after abstracting this

**Review by jamessinmaojun — APPROVED** [review]
lgtm!

**Pull request merged**